### PR TITLE
continuing #839: Extend attrs brain to support provisional APIs

### PR DESCRIPTION
**Pull request by jstriebel**
## Description

This PR continues the original PR #839, which seems to be stale. Only thing missing are formatting and obeying the linter.
From the original PR:
> With attrs 20, new "provisional" APIs were added (see https://www.attrs.org/en/stable/api.html?highlight=field#provisional-apis).
>
> This PR adds support to those new functions in the existing attrs brain.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |

## Related Issue
* related to #656

### CODE REVIEW

**Review by Pierre-Sassoulas — APPROVED** [review]
The coverage is expected to fail don't worry about it coverall is down right now.

**Comment by jstriebel** [comment]
@Pierre-Sassoulas Thanks for the fast review! I adapted the formatting, CI is green now, but needs an additional OK from a maintainer.

File: ChangeLog
Comment (jstriebel) [inline ChangeLog:9]: I'm not sure if this is supposed to go in 2.9, I think it could easily be marked as a patch for semver. I'll move it to 2.8.1 for now.

**Comment by jstriebel** [comment]
@Pierre-Sassoulas Is there anything blocking here? Or could you merge this PR? Thank you!

**Comment by Pierre-Sassoulas** [comment]
Hey @hippo91 could you review this, please ? I'm not confident enough to merge astroid brains change :)

File: astroid/brain/brain_attrs.py
Comment (hippo91) [inline astroid/brain/brain_attrs.py:13]: I do not understand why there is "field" alone? 
By the way i wonder about the existence of `attrib` alone also.

**Review by hippo91 — APPROVED** [review]
@jstriebel thanks for this PR. Everything looks ok. As i understand the new attrs API, `define`, `mutable` and `frozen` are in fact `attr.s` with specialization of the default arguments.
`attrs.field` is an alias of `attrs.ib`. So the modifications proposed here are ok for me even if i wonder about the utility of `field` and `attrib` names without `attrs` prefix.

File: astroid/brain/brain_attrs.py
Comment (jstriebel) [reply astroid/brain/brain_attrs.py:13]: I guess this would help in cases where they are imported as `from attr import attrib` or `from attr import field`. Since this is only used to detect the attributes after an attrs-class is identified, I think the risk of a name collision should be negligible.

**Comment by jstriebel** [comment]
> As i understand the new attrs API, define, mutable and frozen are in fact attr.s with specialization of the default arguments.

Exactly, this only adds some new decorator- and function-names that were added with the next-generation API (indeed with different defaults than before), the main mechanisms stay exactly the same.

File: astroid/brain/brain_attrs.py
Comment (hippo91) [reply astroid/brain/brain_attrs.py:13]: Ok, thanks, that's what i thought too. I also think that there is no risk of name collision.

**Comment by jstriebel** [comment]
@Pierre-Sassoulas @hippo91 Is there anything missing to merge this? Or do you track this via the milestones and merge it later? Sorry to be bothering :pray:

**Comment by Pierre-Sassoulas** [comment]
Thank you for the remainder, let's merge that in 2.8.1 as it's finished earlier than I thought :)

**Pull request merged**
